episodes/filters-and-thresholding.md

```diff
@@ -0,0 +1,743 @@
+exercise](#manual-thresholds). If you don't have that image open, run the top 
+code block in that exercise to open the image:
+
+![](fig/manual-thresholding-exercise-shapes.png){alt="Noisy image containing a 
+rectangle, circle and triangle"}
+
+Try some of the other automatic thresholding options provided by the 
+`napari-segment-blobs-and-things-with-membranes` and 
+`napari-simpleitk-image-processing` plugins.
+
+For example:
+
+- `Tools > Segmentation / binarization > Threshold (Triangle method, Zack et al 1977, scikit-image, nsbatwm)`
+- `Tools > Segmentation / binarization > Threshold (Yen et al 1995, scikit-image, nsbatwm)`
+- `Tools > Segmentation / binarization > Threshold (Li et al 1993, scikit-image, nsbatwm)`
+- `Tools > Segmentation / binarization > Threshold Otsu, multiple thresholds (n-SimpleITK)`
+
+How do they compare to standard Otsu thresholding?  
+`Tools > Segmentation / binarization > Threshold (Otsu et al 1979, scikit-image, nsbatwm)`
+
+:::::::::::::::::::::::: solution 
+
+## Solution
+
+Standard Otsu thresholding chooses a threshold that separates the background 
+from the three shapes:
+
+![](fig/otsu-shapes.png){alt="Mask of shapes (brown) overlaid on shapes image - 
+made with Otsu thresholding"}
+
+Li thresholding gives a very similar result.
+
+Triangle also gives a similar result, but includes some isolated pixels from the 
+noisy background:
+
+![](fig/triangle-shapes.png){alt="Mask of shapes (brown) overlaid on shapes 
+image - made with triangle thresholding"}
+
+Yen gives a different result - isolating the triangle and circle from the rest 
+of the image. Some of the pixels in the rectangle are also labelled, but only 
+in patchy areas:
+
+![](fig/yen-shapes.png){alt="Mask of shapes (brown) overlaid on shapes image - 
+made with Yen thresholding"}
+
+Finally, the 'Threshold Otsu, multiple thresholds' option gives a completely 
+different result. Using 'number of thresholds' of 3, a 'label offset' of 0 and 
+'number of histogram bins' of 256, we create a segmentation where each shape 
+gets its own label:
+
+![](fig/otsu-multiple-shapes.png){alt="Mask of shapes (brown) overlaid on shapes 
+image - made with multiple thresholds Otsu method"}
+
+The 'Threshold Otsu, multiple thresholds' option is an extension of the standard 
+Otsu thresholding method. This allows it to choose multiple thresholds at once 
+(rather than only one as standard). This is useful if you have multiple 
+different classes to segment in your images, that each have a clear peak in the 
+image histogram.
+
+The important point is that different automatic thresholding methods will work 
+well for different kinds of images, and depending on which part of an image you 
+are trying to segment. It's worth trying a variety of options on your images to 
+see which performs best for your specific dataset. 
+
+:::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+## The Napari assistant
```

Comment:
Should talk here about automating and saving the workflow. That seems to be what we can do with the napari assistant. Using the plugin's individually isn't easily repeatable. Setting up a workflow with the assistant and saving it makes the workflow repeatable.